ank/src/cli_commands.rs

```diff
@@ -936,17 +928,20 @@ impl CliCommands {
         };
         output_debug!("Request to run new workload: {:?}", new_workload);
 
-        let res_complete_state = self.get_complete_state(&Vec::new()).await?;
-        output_debug!("Got current state: {:?}", res_complete_state);
-        let mut new_state = *res_complete_state.clone();
-        new_state
+        let mut complete_state_update = CompleteState::default();
+        complete_state_update
             .desired_state
             .workloads
-            .insert(workload_name, new_workload);
+            .insert(workload_name.clone(), new_workload);
 
-        let update_mask = vec!["desiredState".to_string()];
+        let update_mask = vec!["desiredState.workloads.".to_string() + &workload_name];
```

Comment:
Optionally, because this single clone is not a performance issue: But you can avoid the clone by switching the construction of the update mask with the insert operation.

Reply:
I have removed the clone by swapping the lines.